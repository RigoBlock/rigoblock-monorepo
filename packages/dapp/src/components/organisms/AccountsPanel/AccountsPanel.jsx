import './AccountsPanel.scss'
import { connect } from 'react-redux'
import AccountSummary from '../AccountSummary'
import ContentWrapper from '../../molecules/ContentWrapper'
import List from '../List'
import PanelHeader from '../../molecules/PanelHeader'
import PropTypes from 'prop-types'
import React from 'react'

const tooltip = 'List of all accounts'

let AccountsPanel = ({ accounts }) => {
  const values = Object.keys(accounts).map(accNum => ({
    provider: accounts[accNum].provider,
    number: accNum,
    balance: accounts[accNum].balance
  }))
  const header = <PanelHeader title={'Accounts'} tooltip={tooltip} />
  const divider = () => <div className="navbar-divider" />
  return (
    <div className="accounts-panel">
      <ContentWrapper header={header}>
        <List Component={AccountSummary} data={values} Divider={divider} />
      </ContentWrapper>
    </div>
  )
}

AccountsPanel.propTypes = {
  accounts: PropTypes.object.isRequired
}

AccountsPanel = connect(state => {
  console.log(state)
  return {
    accounts: state.user.blockChain.accounts
  }
})(AccountsPanel)

export default AccountsPanel
