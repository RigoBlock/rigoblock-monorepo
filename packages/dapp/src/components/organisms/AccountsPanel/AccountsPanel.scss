@import "../../_settings/variables.scss";

.accounts-panel {
  max-width: 5 * $dist-6xl;

  .navbar-divider {
    width: $dist-l;
    height: 1px;
    margin: 0 $dist-2xl;
    background-color: $dark-grey;
  }

  .account-summary:first-of-type {
    padding-top: 0;
  }
}
