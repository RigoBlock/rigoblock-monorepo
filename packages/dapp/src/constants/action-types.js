import keyMirror from 'keymirror'

export const actionTypes = keyMirror({
  // Global actions
  GLOBAL_INIT: null,
  // Blockchain actions
  BLOCKCHAIN_INIT: null,
  BLOCKCHAIN_ERROR: null,
  LOGGED_IN: null,
  LOGGED_OUT: null
})
