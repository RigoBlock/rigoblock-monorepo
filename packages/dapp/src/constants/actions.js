export default {
  persist: 'persist',
  form: '@@redux-form'
}
