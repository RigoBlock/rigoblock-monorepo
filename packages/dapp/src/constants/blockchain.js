export const VAULT = 'vault'
export const DRAGO = 'drago'
