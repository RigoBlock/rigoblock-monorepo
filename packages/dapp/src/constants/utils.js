import isArray from 'lodash/isArray'
import isObject from 'lodash/isObject'

export const ETHTOMICRO = 1e6
export const ETHTOWEI = 1e18

// we are filtering for BigNumbers
// TODO: remove once isBigNumber() starts working
export const checkBigNumber = val =>
  isObject(val) && Object.keys(val).includes('c') && isArray(val.c)
