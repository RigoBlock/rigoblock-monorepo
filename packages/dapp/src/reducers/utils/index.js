export const composeReducers = (...fns) => (state, action) =>
  fns.reduceRight((arg, f) => f(arg, action), state)
