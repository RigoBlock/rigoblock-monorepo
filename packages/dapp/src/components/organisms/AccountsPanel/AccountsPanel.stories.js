import '../../_settings/_base.scss'
import { Provider } from 'react-redux'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import AccountsPanel from './AccountsPanel'
import React from 'react'
import mockStore from '../../../fixtures/store'

const store = mockStore()

storiesOf('Organisms/AccountsPanel', module)
  .addDecorator(withKnobs)
  .addDecorator(story => <Provider store={store}>{story()}</Provider>)
  .add('default', () => <AccountsPanel />)
