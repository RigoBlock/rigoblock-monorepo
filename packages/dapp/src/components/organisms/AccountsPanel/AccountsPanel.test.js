import { Provider } from 'react-redux'
import { mockStore } from '../../../constants/utils'
import { mount } from 'enzyme'
import AccountsPanel from './AccountsPanel.jsx'
import React from 'react'
import toJson from 'enzyme-to-json'

const store = mockStore(jest.fn)

const wrapper = mount(
  <Provider store={store}>
    <AccountsPanel />
  </Provider>
)

describe('AccountsPanel component', () => {
  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
