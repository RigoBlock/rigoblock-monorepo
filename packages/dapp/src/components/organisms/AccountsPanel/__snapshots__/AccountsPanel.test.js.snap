// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountsPanel component renders correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(AccountsPanel)>
    <AccountsPanel
      accounts={
        Object {
          "0x242b2dd21e7e1a2b2516d0a3a06b58e2d9bf9196": Object {
            "balance": "57999999999960203063",
            "provider": "metamask",
          },
          "0x7328ef1d7ab7583eb9968b2f4a9c900f8a2e2d6d": Object {
            "balance": "67999999999977505798",
            "provider": "metamask",
          },
          "0x8bb7481495d45ccd5cffae1c3a84155fea85a323": Object {
            "balance": "87999999999999410580",
            "provider": "metamask",
          },
        }
      }
      dispatch={[MockFunction]}
    >
      <div
        className="accounts-panel"
      >
        <ContentWrapper
          header={
            <PanelHeader
              title="Accounts"
              tooltip="tooltip"
            />
          }
        >
          <div
            className="content-wrapper"
          >
            <div
              className="content-wrapper-header"
            >
              <PanelHeader
                title="Accounts"
                tooltip="tooltip"
              >
                <div
                  className="panel-header"
                >
                  <h1>
                    Accounts
                  </h1>
                  <Tooltip
                    onClick={[Function]}
                    size="medium"
                    tooltipText=""
                    type="help"
                  >
                    <div
                      className="material-icons icon medium"
                      onClick={[Function]}
                      tooltip=""
                    >
                      help
                    </div>
                  </Tooltip>
                </div>
              </PanelHeader>
            </div>
            <div
              className="content-wrapper-content"
            >
              <List
                Component={[Function]}
                Divider={[Function]}
                className={null}
                data={
                  Array [
                    Object {
                      "balance": "57999999999960203063",
                      "number": "0x242b2dd21e7e1a2b2516d0a3a06b58e2d9bf9196",
                      "provider": "metamask",
                    },
                    Object {
                      "balance": "67999999999977505798",
                      "number": "0x7328ef1d7ab7583eb9968b2f4a9c900f8a2e2d6d",
                      "provider": "metamask",
                    },
                    Object {
                      "balance": "87999999999999410580",
                      "number": "0x8bb7481495d45ccd5cffae1c3a84155fea85a323",
                      "provider": "metamask",
                    },
                  ]
                }
              >
                <div
                  className={null}
                >
                  <AccountSummary
                    balance="57999999999960203063"
                    key="0"
                    number="0x242b2dd21e7e1a2b2516d0a3a06b58e2d9bf9196"
                    provider="metamask"
                    tokenBalance={0}
                  >
                    <div
                      className="account-summary"
                    >
                      <AccountItem
                        appearance="full"
                        number="0x242b2dd21e7e1a2b2516d0a3a06b58e2d9bf9196"
                        provider="metamask"
                      >
                        <div
                          className="account-item-full"
                        >
                          <div
                            className="provider"
                          >
                            <div
                              className="logo"
                            >
                              <Component>
                                SVG image metamask-fox.svg
                              </Component>
                            </div>
                            <span
                              className="name"
                            >
                              MetaMask
                            </span>
                          </div>
                          <span
                            className="account-number"
                          >
                            <span>
                              0x242b2dd21e7e1a2b2516d0a3a06b58e2d9bf9196
                            </span>
                            <Tooltip
                              onClick={[Function]}
                              size="small"
                              tooltipText=""
                              type="filter_none"
                            >
                              <div
                                className="material-icons icon small"
                                onClick={[Function]}
                                tooltip=""
                              >
                                filter_none
                              </div>
                            </Tooltip>
                          </span>
                        </div>
                      </AccountItem>
                      <div
                        className="balances"
                      >
                        <Amount
                          size="small"
                          symbol="ETH"
                          value={57.9999999999602}
                        >
                          <div
                            className="amount small"
                          >
                            <div
                              className="amount-symbol"
                            >
                              ETH
                            </div>
                            <ItemValue
                              currency="ETH"
                              currencyGrowth={0}
                              growth={0}
                              itemValue={57.9999999999602}
                              precision={2}
                              valueSize="small"
                            >
                              <div>
                                <span
                                  className="item-value small"
                                >
                                  58.00
                                </span>
                              </div>
                            </ItemValue>
                          </div>
                        </Amount>
                        <Amount
                          size="small"
                          symbol="GRG"
                          value={0}
                        >
                          <div
                            className="amount small"
                          >
                            <div
                              className="amount-symbol"
                            >
                              GRG
                            </div>
                            <ItemValue
                              currency="ETH"
                              currencyGrowth={0}
                              growth={0}
                              itemValue={0}
                              precision={2}
                              valueSize="small"
                            >
                              <div>
                                <span
                                  className="item-value small"
                                >
                                  0
                                </span>
                              </div>
                            </ItemValue>
                          </div>
                        </Amount>
                      </div>
                    </div>
                  </AccountSummary>
                  <divider
                    key="divider-1"
                  >
                    <div
                      className="navbar-divider"
                    />
                  </divider>
                  <AccountSummary
                    balance="67999999999977505798"
                    key="1"
                    number="0x7328ef1d7ab7583eb9968b2f4a9c900f8a2e2d6d"
                    provider="metamask"
                    tokenBalance={0}
                  >
                    <div
                      className="account-summary"
                    >
                      <AccountItem
                        appearance="full"
                        number="0x7328ef1d7ab7583eb9968b2f4a9c900f8a2e2d6d"
                        provider="metamask"
                      >
                        <div
                          className="account-item-full"
                        >
                          <div
                            className="provider"
                          >
                            <div
                              className="logo"
                            >
                              <Component>
                                SVG image metamask-fox.svg
                              </Component>
                            </div>
                            <span
                              className="name"
                            >
                              MetaMask
                            </span>
                          </div>
                          <span
                            className="account-number"
                          >
                            <span>
                              0x7328ef1d7ab7583eb9968b2f4a9c900f8a2e2d6d
                            </span>
                            <Tooltip
                              onClick={[Function]}
                              size="small"
                              tooltipText=""
                              type="filter_none"
                            >
                              <div
                                className="material-icons icon small"
                                onClick={[Function]}
                                tooltip=""
                              >
                                filter_none
                              </div>
                            </Tooltip>
                          </span>
                        </div>
                      </AccountItem>
                      <div
                        className="balances"
                      >
                        <Amount
                          size="small"
                          symbol="ETH"
                          value={67.9999999999775}
                        >
                          <div
                            className="amount small"
                          >
                            <div
                              className="amount-symbol"
                            >
                              ETH
                            </div>
                            <ItemValue
                              currency="ETH"
                              currencyGrowth={0}
                              growth={0}
                              itemValue={67.9999999999775}
                              precision={2}
                              valueSize="small"
                            >
                              <div>
                                <span
                                  className="item-value small"
                                >
                                  68.00
                                </span>
                              </div>
                            </ItemValue>
                          </div>
                        </Amount>
                        <Amount
                          size="small"
                          symbol="GRG"
                          value={0}
                        >
                          <div
                            className="amount small"
                          >
                            <div
                              className="amount-symbol"
                            >
                              GRG
                            </div>
                            <ItemValue
                              currency="ETH"
                              currencyGrowth={0}
                              growth={0}
                              itemValue={0}
                              precision={2}
                              valueSize="small"
                            >
                              <div>
                                <span
                                  className="item-value small"
                                >
                                  0
                                </span>
                              </div>
                            </ItemValue>
                          </div>
                        </Amount>
                      </div>
                    </div>
                  </AccountSummary>
                  <divider
                    key="divider-2"
                  >
                    <div
                      className="navbar-divider"
                    />
                  </divider>
                  <AccountSummary
                    balance="87999999999999410580"
                    key="2"
                    number="0x8bb7481495d45ccd5cffae1c3a84155fea85a323"
                    provider="metamask"
                    tokenBalance={0}
                  >
                    <div
                      className="account-summary"
                    >
                      <AccountItem
                        appearance="full"
                        number="0x8bb7481495d45ccd5cffae1c3a84155fea85a323"
                        provider="metamask"
                      >
                        <div
                          className="account-item-full"
                        >
                          <div
                            className="provider"
                          >
                            <div
                              className="logo"
                            >
                              <Component>
                                SVG image metamask-fox.svg
                              </Component>
                            </div>
                            <span
                              className="name"
                            >
                              MetaMask
                            </span>
                          </div>
                          <span
                            className="account-number"
                          >
                            <span>
                              0x8bb7481495d45ccd5cffae1c3a84155fea85a323
                            </span>
                            <Tooltip
                              onClick={[Function]}
                              size="small"
                              tooltipText=""
                              type="filter_none"
                            >
                              <div
                                className="material-icons icon small"
                                onClick={[Function]}
                                tooltip=""
                              >
                                filter_none
                              </div>
                            </Tooltip>
                          </span>
                        </div>
                      </AccountItem>
                      <div
                        className="balances"
                      >
                        <Amount
                          size="small"
                          symbol="ETH"
                          value={87.9999999999994}
                        >
                          <div
                            className="amount small"
                          >
                            <div
                              className="amount-symbol"
                            >
                              ETH
                            </div>
                            <ItemValue
                              currency="ETH"
                              currencyGrowth={0}
                              growth={0}
                              itemValue={87.9999999999994}
                              precision={2}
                              valueSize="small"
                            >
                              <div>
                                <span
                                  className="item-value small"
                                >
                                  88.00
                                </span>
                              </div>
                            </ItemValue>
                          </div>
                        </Amount>
                        <Amount
                          size="small"
                          symbol="GRG"
                          value={0}
                        >
                          <div
                            className="amount small"
                          >
                            <div
                              className="amount-symbol"
                            >
                              GRG
                            </div>
                            <ItemValue
                              currency="ETH"
                              currencyGrowth={0}
                              growth={0}
                              itemValue={0}
                              precision={2}
                              valueSize="small"
                            >
                              <div>
                                <span
                                  className="item-value small"
                                >
                                  0
                                </span>
                              </div>
                            </ItemValue>
                          </div>
                        </Amount>
                      </div>
                    </div>
                  </AccountSummary>
                </div>
              </List>
            </div>
          </div>
        </ContentWrapper>
      </div>
    </AccountsPanel>
  </Connect(AccountsPanel)>
</Provider>
`;
